modules/xml/src/smithy4s/xml/internals/XmlDecoder.scala

```diff
@@ -39,11 +40,21 @@ import cats.data.NonEmptyList
 private[smithy4s] trait XmlDecoder[A] { self =>
   def decode(cursor: XmlCursor): Either[XmlDecodeError, A]
   final def map[B](f: A => B): XmlDecoder[B] = new XmlDecoder[B] {
+    override def down(name: XmlQName): XmlDecoder[B] = self.down(name).map(f)
```

Comment:
is this some sort of optimization?

Reply:
Those were artifacts of a solution to a problem that ended up being solved differently. Will remove.

Reply:
[d45899d](https://github.com/disneystreaming/smithy4s/pull/503/commits/d45899dc33844c1b601d57e2076470c2af4b1a94)